### ISSUE TITLE
Invalid references in twisted.cred

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8513]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)| @acabhishek942 reported|
|-|-|
|Trac ID|trac#8513|
|Type|enhancement|
|Created|2016-06-27 15:52:44Z|
|Branch|https://github.com/twisted/twisted/tree/cred-epytext-8513|

Following are the invalid references in twisted.conch obtained by checking out the trunk and running

```
./bin/admin/build-apidocs . apidocs

```

```
twisted.cred.credentials.ICredentials: invalid ref to zope.interface.implementer
twisted.cred.strcred.ICheckerFactory.generateChecker: invalid ref to ICredentialChecker
```

<details><summary>Searchable metadata</summary>

```
trac-id__8513 8513
type__enhancement enhancement
reporter__acabhishek942 acabhishek942
priority__normal normal
milestone__None None
branch__cred_epytext_8513 cred-epytext-8513
branch_author__acabhishek942 acabhishek942
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1467042764275052 1467042764275052
changetime__1467198222005171 1467198222005171
version__None None
owner__acabhishek acabhishek

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_1"></a>|@acabhishek942 commented|
|-|-|

See: https://github.com/twisted/twisted/pull/236

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)|@ralphm set owner to @ralphm<br>@ralphm set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_3"></a>|@ralphm set owner to @acabhishek<br>@ralphm set status to `new`|
|-|-|

Thanks for your PR!

 * The change from `L{}` to `C{}` for `zope.interface.implementer` doesn't seem right to me. We want to use `L{}` to denote that this is a named Python object that should create a link to its documentation. Unfortunately it seems that pydoctor doesn't currently find the target for this reference properly. This can be done by including a `moresystems` directive in the configuration file for pydoctor, which points to a pickle file for docs generated for `zope.interface`. I don't know why that's currently not done, so I will ask around for this. In any case, you should revert this change.
 * The other change for the fully qualified name of `twisted.cred.checkers.ICredentialsChecker` looks good.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban commented|
|-|-|

I have created [#8515](https://github.com/twisted/twisted/issues/8515) to look at letting pydoctor know about zope.interface API

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|GitHub set status to `closed`|
|-|-|

In changeset d4b0935bf18e1f88c55c4b4584c8ca3f0d64748e
```
#!CommitTicketReference repository="" revision="d4b0935bf18e1f88c55c4b4584c8ca3f0d64748e"
Merge 8513-acabhishek942-doc: Fix invalid references in twisted.cred 

Author: acabhishek942
Reviewer: rodrigc
Fixes: #8513
```
